R/fn.R

```diff
@@ -62,6 +62,26 @@ prim_name <- function(prim) {
 #' functions. See [is_function()] for a discussion of primitive and
 #' closure functions.
 #'
+#' Note that the argument names are taken from the closures that are
+#' created when passing the primitive to [as_closure()]. For instance,
+#' while the arguments of the primitive operator `+` are labelled `e1`
+#' and `e2`, `fn_fmls_names()` will return `.x` and `.y`. Note that
+#' for many primitives the base R argument names are purely
+#' placeholders since they don't perform regular argument matching.
+#' E.g. this returns `5` instead of `-5`:
+#'
+#' ```
+#' `-`(e2 = 10, 5)
+#' ```
+#'
+#' To regularise the semantics of primitive functions, it is usually a
+#' good idea to coerce them to a closure first:
```

Comment:
hmm I can't find a meaningful example where this is useful though. That would probably obscure the documentation. Maybe we should issue an error when not a closure.